### PR TITLE
Tiny-world

### PR DESCRIPTION
**Pull request by MarharytaBoichenko**
### A Tiny JS World

[Demo](https://marharytaboichenko.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/MarharytaBoichenko/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/MarharytaBoichenko/tiny-world/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  "species",
+  "gender",
+  "hands",
+  "legs",
+  "saying",
+  "friends",
+];
+
+inhabitants.forEach((item) => {
+  console.log(item);
```
Comment (OleksiyRudenko) [inline submissions/MarharytaBoichenko/tiny-world/index.js:63]: Please remove debugging statements from code here and before submitting other tasks.

File: submissions/MarharytaBoichenko/tiny-world/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  species: "man",
+  gender: "male",
+  hands: 2,
+  legs: 2,
+  saying: "hello",
+  friends: ["Bob", "Ann", "Nick"],
+};
+const woman = {
+  name: "Ann",
+  species: "woman",
```
Comment (OleksiyRudenko) [inline submissions/MarharytaBoichenko/tiny-world/index.js:19]: Men and women belong to the same species.

File: submissions/MarharytaBoichenko/tiny-world/index.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  "legs",
+  "saying",
+  "friends",
+];
+
+inhabitants.forEach((item) => {
+  console.log(item);
+  const allInfoInItem = allProperties.map((prop) =>
+    item[prop] ? item[prop] : `no ${prop}`
+  );
```
Comment (OleksiyRudenko) [inline submissions/MarharytaBoichenko/tiny-world/index.js:66]: You don't really want to spend resource on creation of locally scoped method at every loop iteration. Think of 100k iterations.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@MarharytaBoichenko well done!
A couple of improvements and we are done.

File: submissions/MarharytaBoichenko/tiny-world/index.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  "hands",
+  "legs",
+  "saying",
+  "friends",
+];
+
+inhabitants.forEach((item) => {
+  print(
+    allProperties
+      .map((prop) => (item[prop] ? item[prop] : `no ${prop}`))
```
Comment (OleksiyRudenko) [inline submissions/MarharytaBoichenko/tiny-world/index.js:65]: You don't really need so many parentheses in arrow functions, they do not really help to identify boundaries of code components, especially in one liners.
```
.map((prop) => (item[prop] ? item[prop] : `no ${prop}`))
.map(prop => item[prop] ? item[prop] : `no ${prop}`)
```

**Review by OleksiyRudenko — APPROVED** [review]
@MarharytaBoichenko awesome job!

File: submissions/MarharytaBoichenko/tiny-world/index.js (hunk 4 above)
Comment (MarharytaBoichenko) [reply submissions/MarharytaBoichenko/tiny-world/index.js:65]: exactly)  thanks

**Pull request merged**